substrate/primitives/src/networks.rs

```diff
@@ -52,12 +112,81 @@ pub const COINS: [Coin; 5] = [Coin::Serai, Coin::Bitcoin, Coin::Ether, Coin::Dai
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 pub enum Coin {
   Serai,
+  External(ExternalCoin),
+}
+
+/// The type used to identify external coins.
+#[derive(
+  Clone, Copy, PartialEq, Eq, PartialOrd, Ord, Hash, Debug, Encode, Decode, MaxEncodedLen, TypeInfo,
```

Comment:
The same commentary on manually implementing Encode/Decode/MaxEncodedLen stands as it did for NetworkId. There is the slight caveat we may want to start not at `1` yet `4` or `8`. This allows using the lower three bits for native coins (if we ever add, say, a native USD analogue).

Also, slight nit, can we define ExternalCoin prior to Coin please?